proposals/0351-regex-builder.md

```diff
@@ -1180,6 +1181,108 @@ If we change `OneOrMore(.any)` to `OneOrMore(.any, .possessive)`, matching fails
 
 The desired behavior in this case is `.reluctant`, where the repetition will match as little of the input string as possible. If we use `OneOrMore(.any, .reluctant)`, the code prints expected output `<code>`.
 
+### Anchors and Lookaheads
+
+Anchors are a way to constrain a regex, or part of a regex, to matching particular locations within an input string. Regex builder provides anchors that correspond to regex syntax anchors. Regex builder also provides two types that represent look-ahead assertions — essentially a non-consuming sub-regex that has to match (or not match) before the regex can proceed. 
+
+```swift
+/// A regex component that matches a specific condition at a particular position
+/// in an input string.
+///
+/// You can use anchors to guarantee that a match only occurs at certain points
+/// in an input string, such as at the beginning of the string or at the end of
+/// a line.
+public struct Anchor: RegexComponent {
+  /// An anchor that matches at the start of a line, including the start of
+  /// the input string.
+  ///
+  /// This anchor is equivalent to `^` in regex syntax when the `m` option
+  /// has been enabled or `anchorsMatchLineEndings(true)` has been called.
+  public static var startOfLine: Anchor { get }
+
+  /// An anchor that matches at the end of a line, including at the end of
+  /// the input string.
+  ///
+  /// This anchor is equivalent to `$` in regex syntax when the `m` option
+  /// has been enabled or `anchorsMatchLineEndings(true)` has been called.
+  public static var endOfLine: Anchor { get }
+
+  /// An anchor that matches at a word boundary.
+  ///
+  /// Word boundaries are identified using the Unicode default word boundary
+  /// algorithm by default. To specify a different word boundary algorithm,
+  /// see the `RegexComponent.wordBoundaryKind(_:)` method.
+  ///
+  /// This anchor is equivalent to `\b` in regex syntax.
+  public static var wordBoundary: Anchor { get }
+
+  /// An anchor that matches at the start of the input string.
+  ///
+  /// This anchor is equivalent to `\A` in regex syntax.
+  public static var startOfSubject: Anchor { get }
+
+  /// An anchor that matches at the end of the input string.
+  ///
+  /// This anchor is equivalent to `\z` in regex syntax.
+  public static var endOfSubject: Anchor { get }
+
+  /// An anchor that matches at the end of the input string or at the end of
+  /// the line immediately before the the end of the string.
+  ///
+  /// This anchor is equivalent to `\Z` in regex syntax.
+  public static var endOfSubjectBeforeNewline: Anchor { get }
+
+  /// An anchor that matches at a grapheme cluster boundary.
+  ///
+  /// This anchor is equivalent to `\y` in regex syntax.
+  public static var textSegmentBoundary: Anchor { get }
+
+  /// An anchor that matches at the first position of a match in the input
+  /// string.
+  ///
+  /// This anchor is equivalent to `\y` in regex syntax.
```

Comment:
```suggestion
  /// This anchor is equivalent to `\G` in regex syntax.
```

IIRC

Reply:
`\y` is correct according to @natecook1000